dash-renderer/src/observers/requestedCallbacks.ts

```diff
@@ -45,14 +49,18 @@ import {
     IBlockedCallback
 } from '../types/callbacks';
 
+import wait from './../utils/wait';
+
 import { getPendingCallbacks } from '../utils/callbacks';
 import { IStoreObserverDefinition } from '../StoreObserver';
 
 const observer: IStoreObserverDefinition<IStoreState> = {
-    observer: ({
+    observer: async ({
         dispatch,
         getState
     }) => {
+        await wait(0);
```

Comment:
Adding this little wait apparently has wide-ranging consequences on the timing of various features. 
- the rendering `effect` is now incorrectly timed and causes `isAppReady` to never resolve at initialization as the event is triggered *before* the request is made, update here https://github.com/plotly/dash/pull/1385/files#diff-31f5093f6a582b2369ad2cd8216c4f67R69
- document title is now applied out of order and overridden by clientside callbacks with `document.title = ...`, updated here https://github.com/plotly/dash/pull/1385/files#diff-bd786cd26acb24c396ec355c2ff1c4a3R16
- timing of certain exceptions seems different enough to cause ordering issues in devtools, updated test here https://github.com/plotly/dash/pull/1385/files#diff-7b6da2b5053a169045af0651bfcb1b80R680

This list is probably not exhaustive yet. Some proposed fixes should make the overall structure more robust, others are patches or small fixes..